src/utils/determineWin.tsx

```diff
@@ -0,0 +1,24 @@
+export function determineWin(board: string[]) {
+  const possibleWins: number[][] = [
+    [0, 1, 2],
+    [3, 4, 5],
+    [6, 7, 8],
+    [0, 3, 6],
+    [1, 4, 7],
+    [2, 5, 8],
+    [0, 4, 8],
+    [2, 4, 6],
+  ]
+
+  let winningLetter = ''
+
+  for (let i = 0; i < possibleWins.length; i++) {
+    const [a, b, c] = possibleWins[i]
+    if (board[a] && board[a] === board[b] && board[a] === board[c]) {
+      winningLetter = board[a]
+      return winningLetter
+    }
+  }
```

Comment:
Nice approach to this problem! You could also make this more declarative by avoiding a for loop. tbh this took me a lot of thinking to figure out but in the end I think it makes sense. We are _reducing_ a 2D array of possibleWins to a string of '' | 'X' | 'O'
```suggestion
  return possibleWins.reduce(
    (acc, [a, b, c]) =>
      board[a] && board[a] === board[b] && board[a] === board[c]
        ? board[a]
        : acc,
    ''
  )
```